device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClient.java

```diff
@@ -725,4 +726,14 @@ public void setOption(String optionName, Object value)
 
     }
 
+    /**
+     * Registers a callback to be executed whenever the connection to the device is lost or established.
+     * 
+     * @param callback the callback to be called.
+     */
+    public void registerConnectionStateCallback(IotHubConnectionStateCallback callback) {
```

Comment:
should null value of call back be accepted ? perhaps throw IllegalArgument Exception in such cases.